app/screens/ProfileScreen.tsx

```diff
@@ -20,7 +24,27 @@ export const ProfileScreen: FC<ProfileScreenProps> = function ProfileScreen(_pro
   return (
     <Screen preset="scroll" contentContainerStyle={$styles.barContainer} safeAreaEdges={["top"]}>
       <BackHeader navigation={navigation} title="Profile" onSavePress={handleSavePress} />
-      <BodyText variant="body4">{profile ? JSON.stringify(profile) : "Loading..."}</BodyText>
+      {isLoading && (
+        <YStack alignItems="center" justifyContent="center" py="$6" gap="$3">
+          <Spinner size="large" />
+          <BodyText variant="body" opacity={0.7}>
+            Loading your profile...
+          </BodyText>
+        </YStack>
+      )}
+
+      {!isLoading && error && (
+        <YStack px="$3" py="$4">
+          <BodyText variant="body">Couldn’t load your profile</BodyText>
+          <BodyText variant="body2" opacity={0.7}>
+            {error instanceof Error ? error.message : "Please try again."}
+          </BodyText>
+        </YStack>
+      )}
+
+      <YStack justifyContent="center" alignItems="center" margin={20} marginBlockStart={40}>
+        <ProfileInfoCard profile={profile} />
+      </YStack>
```

Comment:
I believe with Tanstack, one of either `isLoading` `error` and `data` is always defined, so I think I would personally opt for something like: 

```suggestion
    {profile && (
        <YStack justifyContent="center" alignItems="center" margin={20} marginBlockStart={40}>
          <ProfileInfoCard profile={profile} />
        </YStack>
      )}
```

I think right now if it errors, it shows BOTH your error state AND an empty profile, which is justifiable as an error state, just confirming that this was your intention.

If you do the `profile &&` then your component doesnt need to consider `| undefined` and less optional chaining which might be cleaner from a dev pov!